File: examples/nodejs/publisher.js
const redis = require('redis');

async function publisher() {
    const client = redis.createClient();
    
    client.on('error', (err) => {
        console.error('Redis Client Error:', err);
    });
    
    try {
        await client.connect();
        console.log('Publisher started. Publishing news updates...');
        
        let count = 1;
        setInterval(async () => {
            const message = `Breaking News #${count}: Important update at ${new Date().toLocaleTimeString()}`;
            try {
                await client.publish('news-channel', message);
                console.log(`Published: ${message}`);
            } catch (err) {
                console.error('Error publishing message:', err);
            }
            count++;
        }, 3000);
    } catch (err) {
        console.error('Failed to connect to Redis:', err);
        process.exit(1);
    }
}

publisher().catch((err) => {
    console.error('Publisher error:', err);
    process.exit(1);
});

File: examples/python/publisher.py
import redis
import time
import datetime

def publisher():
    """Publisher that sends messages to the news-channel"""
    client = redis.Redis(host='localhost', port=6379, decode_responses=True)
    print('Publisher started. Publishing news updates...')
    
    count = 1
    while True:
        message = f"Breaking News #{count}: Important update at {datetime.datetime.now().strftime('%H:%M:%S')}"
        client.publish('news-channel', message)
        print(f'Published: {message}')
        count += 1
        time.sleep(3)

if __name__ == '__main__':
    try:
        publisher()
    except KeyboardInterrupt:
        print('\nPublisher stopped.')
    except Exception as e:
        print(f'Error: {e}')
